### ISSUE TITLE
What New Feature should we add on this API ?

### ISSUE THREAD

**Issue by dasmond1** [salman0ansari/whatsapp-api-nodejs#56]
If You Have any idea, to add new feature on this API. Leave a comment below.

**Comment by salman0ansari** [comment]
we need more group endpoint

**Comment by salman0ansari** [comment]
but before anything we need to implement data store

**Comment by dasmond1** [comment]
> but before anything we need to implement data store

mongo or mysql ? what your opinion ?

**Comment by salman0ansari** [comment]
mongo would be good

**Comment by manojsitapara** [comment]
Add endpoint to set webhook and remove webhook

**Comment by melvas6** [comment]
Message Delivery status will be very helpful

**Comment by Masterxv** [comment]
i needed image with template button

**Comment by salman0ansari** [comment]
> i needed image with template button

Already added please check before opening new issue

**Comment by dasmond1** [comment]
For Group Method, i will simplfy the code letter according to PR #63

**Comment by jazz7381** [comment]
socket.io implementation

**Comment by salman0ansari** [comment]
> socket.io implementation

why?

**Comment by billantony** [comment]
implementar en webhook la posibilidad de obtener los mensajes enviados desde los dispositivos

**Comment by salman0ansari** [comment]
> implement in webhook the possibility of obtaining the messages sent from the devices

english pls

**Comment by billantony** [comment]
implement in webhook the possibility of obtaining the messages sent from the devices

**Comment by dasmond1** [comment]
> > implement in webhook the possibility of obtaining the messages sent from the devices
> 
> 
> 
> What ?? 😕

I know what he want. But it's take time to implement with socket.io

Just be patient and stars this project to get new notification and update.

**Comment by salman0ansari** [comment]
> > > implement in webhook the possibility of obtaining the messages sent from the devices
> > 
> > 
> > What ?? 😕
> 
> I know what he want. But it's take time to implement with socket.io
> 
> Just be patient and stars this project to get new notification and update.

he want socket.io implementation?

**Comment by dasmond1** [comment]
> > > > implement in webhook the possibility of obtaining the messages sent from the devices
> > > 
> > > 
> > > What ?? 😕
> > 
> > 
> > I know what he want. But it's take time to implement with socket.io
> > Just be patient and stars this project to get new notification and update.
> 
> he want socket.io implementation?

i think yes 😃 . But i will try develop in my private branch. if is good for you i will make a PR. 'cause i have a code from my friend that is using Baileys to and using socket.io for pushing a respond to client.

**Comment by salman0ansari** [comment]
> > > > > implement in webhook the possibility of obtaining the messages sent from the devices
> > > > 
> > > > 
> > > > What ?? 😕
> > > 
> > > 
> > > I know what he want. But it's take time to implement with socket.io
> > > Just be patient and stars this project to get new notification and update.
> > 
> > 
> > he want socket.io implementation?
> 
> i think yes 😃 . But i will try develop in my private branch. if is good for you i will make a PR. 'cause i have a code from my friend that is using Baileys to and using socket.io for pushing a respond to client.

this project only foucs on REST implement if you want to implement socket.io for that we can create a different branch.

**Comment by sairmali** [comment]
1. *** we can make unlimited scan numbers 
2. add auto registration without scan phone throw whatsapp channel (auto wart whatsapp registration)

i have lots of ideas

**Comment by salman0ansari** [comment]
> 1. *** we can make unlimited scan numbers
> 2. add auto registration without scan phone throw whatsapp channel (auto wart whatsapp registration)
> 
> i have lots of ideas

1. It's technically impossible to scan unlimited numbers
2. QRCode is the only way you can connect to WhatsApp Web.

**Comment by akbardwi** [comment]
i think add a custom webhook url per instance is good

**Comment by fitwan** [comment]
Please, Send Location is good

**Comment by shitross** [comment]
sticker support, for example we can decrypt an image and convert it to a sticker

**Comment by Zarinia** [comment]
support docker compose

**Comment by sairmali** [comment]
we can add messages forward options?

**Comment by dasmond1** [comment]
> sticker support, for example we can decrypt an image and convert it to a sticker

working on it 👍🏻

**Comment by Zarinia** [comment]
Delete message from chat or group

**Comment by 5569866** [comment]
Hello, we can add proxy socket5? Better support multiple users and prevent WhatsApp from being prohibited

**Comment by salman0ansari** [comment]
> Hello, we can add proxy socket5? Better support multiple users and prevent WhatsApp from being prohibited

Nope, people will misuse it if you want to add something to your like you can happily fork repo

**Comment by hdmidia** [comment]
Hi, first of all, a thank you... The API is great better than some I've tested.

I'm missing a route /close or /logoff to close the session without deleting it... I tried to do it here but without success.

`exports.logoff = async (req, res, next) => {`
`    let errormsg;`
`    try {`
`        await WhatsAppInstances[req.query.key].instance?.sock?.logoff();`
`    } catch (error) {`
`        errormsg = error;`
`    }`
`    return res.json({`
`        error: false,`
`        message: 'logoff successfull',`
`        errormsg: errormsg ? errormsg : null,`
`    });`
`};`

I'm learning to use Baileys now, I hope to contribute to the community soon.

**Comment by Sulivan1999** [comment]
> but before anything we need to implement data store

Hello. When do you intend to make this storage available through the database?

**Comment by sairmali** [comment]
@salman0ansari @achdiadsyah @akbardwi  @manojsitapara  we can add messages forward option

**Comment by nestordavalos** [comment]
congratulations for the project is very good and very easy to implement, you could add location, vcard and reactionMessage

**Comment by jazz7381** [comment]
how abot using token in header as bearer intead of query params?
